setup.py

```diff
@@ -230,7 +232,11 @@ def run(self):
 
         # Do we actually need this here?
         if WITH_NNPACK:
-            print('-- Detected NNPACK at ' + NNPACK_LIB_DIR)
+            if NNPACK_IS_USER_DEFINED:
+                nnpack_dir = NNPACK_LIB_DIR
+            else:
+                nnpack_dir = NNPACK_LIB_PATHS[0]
```

Comment:
Code like this always makes me nervous (what if it's empty? What about the rest of it?) but I guess you had a good reason for doing it this way...

Reply:
This should also be inside the tool. Why does setup.py need to know what’s the difference between NNPACK_LIB_DIR and NNPACK_LIB_PATHS. to me it seems that they’re pretty much the same thing

Reply:
@ezyang its sort of an invariant that to hit this code path this list should not be empty.